service-base/src/main/java/org/eclipse/hono/service/AbstractServiceBase.java

```diff
@@ -0,0 +1,182 @@
+package org.eclipse.hono.service;
+
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Future;
+import org.eclipse.hono.config.HonoConfigProperties;
+import org.eclipse.hono.util.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+public abstract class AbstractServiceBase extends AbstractVerticle {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractServiceBase.class);
+
+    protected HonoConfigProperties config = new HonoConfigProperties();
+
+    protected int port         = Constants.PORT_UNCONFIGURED;
+    protected int insecurePort = Constants.PORT_UNCONFIGURED;
+
+    /**
+     * Define the default value for the TLS port number in the implementing server class (e.g. 5671 for AMQPS 1.0).
+     * @return The default value for the TLS port.
+     */
+    protected abstract int getPortDefaultValue();
+    /**
+     * Define the default value for the unencrypted port number in the implementing server class (e.g. 5672 for AMQP 1.0).
+     * @return The default value for the unencrypted port.
+     */
+    protected abstract int getInsecurePortDefaultValue();
+
+    /**
+     * Return the port number of the TLS port. If the implementing service determined the port during startup
+     * (ephemeral port configuration), return this port number.
+     * @return The port number of the TLS port.
+     */
+    public abstract int getPort();
+
+    /**
+     * Return the port number of the unencrypted port. If the implementing service determined the port during startup
+     * (ephemeral port configuration), return this port number.
+     * @return The port number of the unencrypted port.
+     */
+    public abstract int getInsecurePort();
+
+    /**
+     * Set the configuration for the Hono service.
+     * @param props The configuration properties.
+     */
+    public final void setConfig(final HonoConfigProperties props) {
```

Comment:
based on what I mentioned above, IMHO this should be the place where the configuration gets injected, thus this method should be annotated with `@Autowired` instead of the server specific method.